### ISSUE TITLE
`MergeDirs` for combine remote

### ISSUE THREAD

**Issue by martinwang2002** [rclone/rclone#6999]
#### What is your current rclone version (output from `rclone version`)?
```
rclone v1.63.0-beta.7008.61d6f538b
- os/version: ubuntu 22.04 (64 bit)
- os/kernel: 5.15.0-70-generic (aarch64)
- os/type: linux
- os/arch: arm64 (ARMv8 compatible)
- go/version: go1.20.4
- go/linking: static
- go/tags: none
```

#### What problem are you are trying to solve?
When deduping folders with combine + google drive,
```
Failed to dedupe: combine root 'path_to_folder': can't merge directories
```

#### How do you think rclone should be changed to solve that?
Similar to https://github.com/rclone/rclone/issues/2689#issuecomment-433222361, we may need `MergeDirs` for combine.

For example, in the crypt remote,

https://github.com/rclone/rclone/blob/fcf47a83934a6f44f44102201880b9a24ebec5d6/backend/crypt/crypt.go#L747-L757

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by martinwang2002** [comment]
I have tested dedupe and it works.

Do I also need to check other methods or functionality?

**Comment by ncw** [comment]
Thanks for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.63

The integration tests test all the other methods so hopefully they are all working now.
